gui/packages/desktop/src/main/index.js

```diff
@@ -295,18 +305,75 @@ const ApplicationMain = {
     this._daemonRpc.connect({ path: DAEMON_RPC_PATH });
   },
 
-  _onStateChange(newState: TunnelStateTransition) {
+  _reconnectToDaemon() {
+    this._reconnectBackoff.attempt(() => {
+      this._connectToDaemon();
+    });
+  },
+
+  _recoverFromBootstrapError(_error: ?Error) {
+    // Attempt to reconnect to daemon if the program fails to fetch settings, tunnel state or
+    // subscribe for RPC events.
+    this._daemonRpc.disconnect();
```

Comment:
FYI very simple recovery from receiving the initial state from the daemon: reconnect and start over.